ci/build_cpp.sh

```diff
@@ -26,33 +26,13 @@ export RAPIDS_ARTIFACTS_DIR
 source rapids-rattler-channel-string
 
 # Construct the extra variants according to the architecture
-if [[ "$(arch)" == "x86_64" ]]; then
-    cat > variants.yaml << EOF
-    c_compiler_version:
-      - 13
-
-    cxx_compiler_version:
-      - 13
-
-    cuda_version:
-      - ${RAPIDS_CUDA_VERSION}
+cat > variants.yaml << EOF
+cuda_version:
+  - ${RAPIDS_CUDA_VERSION}
 EOF
-else
-    cat > variants.yaml << EOF
-    zip_keys:
-    - [c_compiler_version, cxx_compiler_version, cuda_version]
-
-    c_compiler_version:
-    - 12
-    - 13
-
-    cxx_compiler_version:
-    - 12
-    - 13
-
-    cuda_version:
-    - 12.1 # The last version to not support cufile
-    - ${RAPIDS_CUDA_VERSION}
+if [[ "$(arch)" == "aarch64" ]]; then
+  cat >> variants.yaml << EOF
+  - 12.1 # The last version to not support cufile
```

Comment:
Then the legacy CUDA ARM builds are appended to the file

This also makes it easier to drop this section of code when it no longer applies